libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/overlay/LiveRideOverlay.java

```diff
@@ -172,15 +238,123 @@ private String speed() {
     return formatter_.speed(location.getSpeed());
   }
 
-  private String distanceUntilTurn() {
+  private int distanceUntilTurn() {
+    final GeoPoint whereIam = whereIAm();
+    final int fromEnd;
+    if (whereIam == null) {
+       fromEnd = Integer.MAX_VALUE;}
+    else {
+      // Get distance to end of active segment
+      final Segment activeSeg = Route.journey().activeSegment();
+      fromEnd = activeSeg.distanceFromEnd(whereIam);
+    }
+    return fromEnd;
+  }
+
+  private GeoPoint whereIAm() {
     final Location location = lastLocation();
     if (location == null)
+      return null;
+    return new GeoPoint(location);
+  }
+
+  private void drawRemainingJourney(Canvas canvas, Rect box) {
+    String strRemDistance;
+    String strRemTime;
+    String ETA = "";
+    int ETAWidth=0;
+    int ETATextWidth=0;
```

Comment:
Please take more care with your formatting generally - for example, we should always have spaces around `=` signs.